ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -184,19 +183,17 @@ public boolean accept(Path path) {
   public static final int MAX_STATEMENTS_PER_TXN = 10000;
   public static final Pattern LEGACY_BUCKET_DIGIT_PATTERN = Pattern.compile("^[0-9]{6}");
   public static final Pattern BUCKET_PATTERN = Pattern.compile("bucket_([0-9]+)(_[0-9]+)?$");
-  private static final Set<Integer> READ_TXN_TOKENS = new HashSet<Integer>();
+  private static final Set<Integer> READ_TXN_TOKENS = new HashSet<>();
 
   private static Cache<String, DirInfoValue> dirCache;
   private static AtomicBoolean dirCacheInited = new AtomicBoolean();
 
   static {
     READ_TXN_TOKENS.addAll(Arrays.asList(
-            HiveParser.TOK_DESCDATABASE,
-            HiveParser.TOK_DESCTABLE,
-            HiveParser.TOK_SHOWTABLES,
-            HiveParser.TOK_SHOW_TABLESTATUS,
-            HiveParser.TOK_SHOW_TBLPROPERTIES,
```

Comment:
Why did we remove show tables tok

Reply:
Because those don't even need a txn. I would have removed all of the operations (EXPLAIN, DESC), however, that cause number of old tests to fail. I'll handle them separately in diff JIRAs 1-by-1,.